em-app/src/platform/sgx.rs

```diff
@@ -42,14 +42,10 @@ pub(crate) fn get_remote_attestation_parameters(
     }.to_der().into();
 
     let attributes = vec![(oid::attestationInlineSgxLocal.clone(), vec![attestation])];
-    let extensions = alt_names.and_then(|names| {
-        Some(vec![(pkix::oid::subjectAltName.clone(), false, pkix::yasna::construct_der(|w| DnsAltNames { names }.write(w)).into())])
-    });
-
+    let extensions = get_extensions_from_alt_names(alt_names);
     let mut sgx_name = SgxName::from_report(&report, true);
-    sgx_name.append(vec![(pkix::oid::commonName.clone(), common_name.to_string().into())]);
-    let subject = sgx_name.to_name();
-    
+    let mut subject = subject.clone(); // Copy the subject, since it must be modified
```

Comment:
consider passing a mut reference of `subject`

Reply:
The nitro code does not need mutablity, so in wanting to keep things as restrained as possible, I tried to avoid using `mut`. The signature of this function does already have `signer: &mut dyn CsrSigner,`, so there is a mutable parameter. Another option is to use a pass by value/copy. The advantage of a mutable parameter would be the user having the modified Subject available with all values, but I do not know if this adds much.